src/Movement.hs

```diff
@@ -0,0 +1,83 @@
+module Movement(performMove) where
+import Data.Array
+import Data.List
+import Game
+import Utils
+import Random
+import GameOver
+
+
+-- Shifts all the non empty cells in a row to the left
+-- Empty cells are moved to the right side of the row
+shiftRow :: [Cell] -> [Cell]
+shiftRow row = filtered ++ padding
+  where
+    filtered = filter (/= Empty) row
```

Comment:
• No instance for ‘Eq Cell’ arising from a use of ‘/=’
• In the expression: (/=)
  In the first argument of ‘filter’, namely ‘(/= Empty)’
  In the expression: filter (/= Empty) rowtypecheck(-Wdeferred-type-errors)